frontend/lib/src/components/elements/Spinner/Spinner.test.tsx

```diff
@@ -78,4 +78,37 @@ describe("Spinner component", () => {
     expect(spinnerContainer).toHaveClass("stCacheSpinner")
     expect(spinnerContainer).toHaveStyle("paddingBottom: 1rem")
   })
+
+  it("shows timer when showElapsedTime is true", () => {
+    render(
+      <BaseProvider theme={LightTheme}>
+        <Spinner {...getProps({}, { showElapsedTime: true })} />
+      </BaseProvider>
+    )
+
+    const spinnerContainer = screen.getByTestId("stSpinner")
+    expect(spinnerContainer).toBeInTheDocument()
+    expect(screen.getByText("(0.0 seconds)")).toBeInTheDocument()
+  })
+})
+
+describe("formatTime", () => {
+  it("formats seconds only", () => {
+    expect(formatTime(0)).toBe("(0.0 seconds)")
+    expect(formatTime(1.5)).toBe("(1.5 seconds)")
+    expect(formatTime(45.2)).toBe("(45.2 seconds)")
+  })
+
+  it("formats minutes and seconds", () => {
+    expect(formatTime(60)).toBe("(1 minute)")
+    expect(formatTime(61.5)).toBe("(1 minute, 1.5 seconds)")
+    expect(formatTime(122.2)).toBe("(2 minutes, 2.2 seconds)")
+  })
+
+  it("formats hours, minutes and seconds", () => {
+    expect(formatTime(3600)).toBe("(1 hour)")
+    expect(formatTime(3660)).toBe("(1 hour, 1 minute)")
+    expect(formatTime(3661.5)).toBe("(1 hour, 1 minute, 1.5 seconds)")
+    expect(formatTime(7384.2)).toBe("(2 hours, 3 minutes, 4.2 seconds)")
+  })
```

Comment:
nit: its better/preferred to use parameterized tests in this case, e.g.:

https://github.com/streamlit/streamlit/blob/27fbf5eaee4c23d56f2b17bc10b5a8ee8ec941f6/frontend/lib/src/components/widgets/DataFrame/columns/utils.test.ts#L240-L260

Reply:
Ah sweet, thx for the tip!